botocore/handlers.py

```diff
@@ -275,8 +276,16 @@ def signature_overrides(service_data, service_name, session, **kwargs):
         logger.debug("Switching signature version for service %s "
                      "to version %s based on config file override.",
                      service_name, signature_version_override)
-        service_metadata = service_data['metadata']
-        service_metadata['signatureVersion'] = signature_version_override
+        return signature_version_override
+
+
+def disable_signing(**kwargs):
+    """
+    This handler disables request signing by setting the signer
+    name to an empty string, similar to how the signature
+    overrides above work.
+    """
+    return ''
```

Comment:
I think I made a change such that disable signing is by specifying ``None``. Default signing is then specified by ``NOT_SET`` which is a global object specified in ``endpoint.py``